src/chainer.test.js

```diff
@@ -1 +1,31 @@
+'use strict';
+
 const { chainer } = require('./chainer');
+
+describe('chainer', () => {
+  it('should chain the unary functions correctly', () => {
+    const mockFunctionA = jest.fn((x) => x + 1);
+    const mockFunctionB = jest.fn((x) => x * 2);
+    const mockFunctionC = jest.fn((x) => x - 3);
+    const mockFunctionD = jest.fn((x) => x / 4);
+
+    const chain = chainer([mockFunctionA, mockFunctionB,
+      mockFunctionC, mockFunctionD]);
+
+    const input = 10;
+    const result = chain(input);
+
+    expect(mockFunctionA).toHaveBeenCalledWith(input);
+    expect(mockFunctionB).toHaveBeenCalledWith(mockFunctionA(input));
+
+    expect(mockFunctionC)
+      .toHaveBeenCalledWith(mockFunctionB(mockFunctionA(input)));
+
+    expect(mockFunctionD)
+      .toHaveBeenCalledWith(mockFunctionC(mockFunctionB(mockFunctionA(input))));
+
+    expect(result)
```

Comment:
The final equality check is overly complex. It would be simpler and more readable to calculate the expected result ahead of time and compare to that.